gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/inmemory/IntegrationAgentInMemory.java

```diff
@@ -60,13 +60,16 @@ public Single<IntegrationSubscription> subscribe(
                 return subscriptionIds;
             }
         );
+        IntegrationSubscription.Type type;
+        if (subscriptionParameter instanceof SubscriptionParameter.ApiKey) {
+            type = IntegrationSubscription.Type.API_KEY;
+        } else if (subscriptionParameter instanceof SubscriptionParameter.OAuth) {
+            type = IntegrationSubscription.Type.OAUTH2;
+        } else {
+            throw new IllegalArgumentException("Unsupported subscription parameter: " + subscriptionParameter);
```

Comment:
You should throw an IntegrationSubscriptionException here to match the real implementation no ?

Reply:
well seen